### PR TITLE
adjust videodir path for mobile servers to accommodate their inabilit…

### PR DESCRIPTION
**Pull request by benptc**
…y to mkdir a videos directory in the WORLD_local path

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1219,15 +1217,20 @@ function objectWebServer() {
         }
 
         if ((urlArray[urlArray.length - 2] === 'videos') && urlArray[urlArray.length - 1].split('.').pop() === 'mp4') {
-            urlArray[urlArray.length - 2] = identityFolderName + '/videos';
+            // videoDir differs on mobile due to inability to call mkdir
+            if (!isMobile) {
+                urlArray[urlArray.length - 2] = identityFolderName + '/videos';
+            } else {
+                urlArray.splice(urlArray.length - 2, 1);
+            }
```
Comment (benptc) [inline server.js:1225]: I acknowledge that this is pretty messy, but I'm struggling to find a cleaner solution given how messy the surrounding implementation is

Comment (hobinjk-ptc) [reply server.js:1225]: Yeah this function is a bit on the opaque side

**Comment by benptc** [comment]
I thought this fixed it entirely but it occasionally crashes when trying to save the video... after 2 hours of intense debugging I have another version that is slightly more stable but still frustratingly not perfect... I'll keep working on this

**Comment by benptc** [comment]
Resolved this mostly on the Xcode side, with a few cleanups and error handling added here

**Pull request merged**